conduit/plugins/importers/algod/algod_importer_test.go

```diff
@@ -69,20 +72,134 @@ func TestInitSuccess(t *testing.T) {
+			[]string{}},
+		{"catchpoint round before node round skips fast catchup", "1234#abcd",
+			NewAlgodServer(
+				GenesisResponder,
+				MakeSyncRoundResponder(200),
+				MakeNodeStatusResponder(models.NodeStatus{LastRound: 1235})),
+			"",
+			[]string{"Skipping catchpoint catchup for 1234#abcd, since it's before node round 1235"}},
+		{"start catchpoint catchup failure", "1236#abcd",
+			NewAlgodServer(
+				GenesisResponder,
+				MakeSyncRoundResponder(200),
+				MakeNodeStatusResponder(models.NodeStatus{LastRound: 1235}),
+				MakeStatusResponder("/v2/catchup/", 400, "")),
+			"POST /v2/catchup/1236#abcd received unexpected error: HTTP 400",
+			[]string{}},
+		{"monitor catchup node status failure", "1236#abcd",
+			NewAlgodServer(
+				GenesisResponder,
+				MakeSyncRoundResponder(200),
+				MakeJsonResponderSeries("/v2/status", []int{200, 400}, []interface{}{models.NodeStatus{LastRound: 1235}}),
+				MakeStatusResponder("/v2/catchup/", 200, "")),
+			"received unexpected error getting node status: HTTP 400",
+			[]string{}},
+		{"monitor catchup success", "1236#abcd",
+			NewAlgodServer(
+				GenesisResponder,
+				MakeSyncRoundResponder(200),
+				MakeJsonResponderSeries("/v2/status", []int{200}, []interface{}{
+					models.NodeStatus{LastRound: 1235},
+					models.NodeStatus{Catchpoint: "1236#abcd", CatchpointProcessedAccounts: 1, CatchpointTotalAccounts: 1},
+					models.NodeStatus{Catchpoint: "1236#abcd", CatchpointVerifiedAccounts: 1, CatchpointTotalAccounts: 1},
+					models.NodeStatus{Catchpoint: "1236#abcd", CatchpointAcquiredBlocks: 1, CatchpointTotalBlocks: 1},
+					models.NodeStatus{Catchpoint: "1236#abcd"},
+					models.NodeStatus{LastRound: 1236},
+				}),
+				MakeStatusResponder("/v2/catchup/", 200, "")),
+			"",
+			[]string{
+				"catchup phase 1 of 4 (Processed Accounts): 1 / 1",
+				"catchup phase 2 of 4 (Verified Accounts): 1 / 1",
+				"catchup phase 3 of 4 (Acquired Blocks): 1 / 1",
+				"catchup phase 4 of 4 (Verified Blocks)",
+			}},
+		{"wait for node to catchup error", "1236#abcd",
+			NewAlgodServer(
+				GenesisResponder,
+				MakeSyncRoundResponder(200),
+				MakeJsonResponderSeries("/v2/status", []int{200, 200, 400}, []interface{}{models.NodeStatus{LastRound: 1235}}),
+				MakeStatusResponder("/v2/catchup/", 200, "")),
+			"received unexpected error (StatusAfterBlock) waiting for node to catchup: HTTP 400",
+			[]string{}},
+	}
+	for _, ttest := range tests {
+		t.Run(ttest.name, func(t *testing.T) {
+			testLogger, hook := test.NewNullLogger()
+			testImporter := New()
+			cfgStr := fmt.Sprintf(`---
+mode: %s
+netaddr: %s
+catchup-config:
+  catchpoint: %s
+`, "follower", ttest.algodServer.URL, ttest.catchpoint)
+			_, err := testImporter.Init(ctx, conduit.MakePipelineInitProvider(&pRound, nil), plugins.MakePluginConfig(cfgStr), testLogger)
+			if ttest.err != "" {
+				assert.ErrorContains(t, err, ttest.err)
+			} else {
+				assert.NoError(t, err)
+			}
```

Comment:
Is there any point checking logs if there was an unexpected error condition?
```suggestion
			if ttest.err != "" {
				require.ErrorContains(t, err, ttest.err)
			} else {
				require.NoError(t, err)
			}
```